### PR TITLE
Simplify setup.py

### PR DESCRIPTION
**Pull request by henryiii**
Dropping lots of custom code. Now requires setuptools.

### CODE REVIEW

**Comment by eduardo-rodrigues** [comment]
@henryiii, did you think about this? We could try and get this merge shortly.

**Comment by henryiii** [comment]
I think it's ready for review.

File: setup.py [hunk 1]
```diff
@@ -157,28 +106,6 @@ def finalize_options(self):
         _install.finalize_options(self)
 
 
-# Only add numpy to *_requires lists if not already installed to prevent
```
[the remark this answers is not in the document]
Comment (eduardo-rodrigues) [reply setup.py]: OK, fine, actually.

[the remark this answers is not in the document]
Comment (henryiii) [reply setup.py]: This was a workaround for distutils, and wasn't needed for setuptools.

**Pull request merged**